### PR TITLE
- Erase operation (hotkey: DELETE)

### PR DESCRIPTION
**Pull request by DjWarmonger**
- New action Erase with hotkey DELETE
- It is possible to undo erase object

However, this one seems to be much more complex under the hood. More work needed.

### CODE REVIEW

File: lib/mapping/CMap.h [hunk 1]
```diff
@@ -488,5 +490,6 @@ class DLL_LINKAGE CMap : public CMapHeader
 		h & CGTownInstance::universitySkills;
 
 		h & instanceNames;
+		h & uid_counter;
```
Comment (Nordsoft91) [inline lib/mapping/CMap.h:493]: could you please increment SERIALIZATION_VERSION from CSerializer.h?

File: lib/mapping/CMapEditManager.cpp [hunk 2]
```diff
@@ -331,21 +343,31 @@ void CComposedOperation::execute()
 
 void CComposedOperation::undo()
 {
-	for(auto & operation : operations)
+	//reverse order
+	for(auto & operation = operations.rbegin(); operation != operations.rend(); operation++)
```
Comment (Nordsoft91) [inline lib/mapping/CMapEditManager.cpp:347]: There is compilation error pointing to this line (see actions). Maybe removing & will help

File: mapeditor/mapcontroller.cpp [hunk 3]
```diff
@@ -143,7 +143,9 @@ void MapController::commitObjectErase(int level)
 	}
 	else if (selectedObjects.size() == 1)
 	{
-		_map->getEditManager()->removeObject(*selectedObjects.begin());
+		_map->getEditManager()->removeObject(selectedObjects.front());
+		//invalidate tiles under object
+		_mapHandler->invalidate(_mapHandler->geTilesUnderObject(selectedObjects.front()));
 	}
```
Comment (DjWarmonger) [inline mapeditor/mapcontroller.cpp:149]: What the heck happened here. This didn't even compile as set has no front() method. Also, this version doesn't work correctly either.

Comment (Nordsoft91) [reply mapeditor/mapcontroller.cpp:149]: Oops, it's my fault- was trying to resolve merge conflicts

File: lib/mapping/CMapEditManager.cpp (hunk 2 above)
Comment (DjWarmonger) [reply lib/mapping/CMapEditManager.cpp]: done

**Pull request merged**